### ISSUE TITLE
Placeholders not selected in snippets for zig zls

### ISSUE THREAD

**Issue by lobre** [kak-lsp/kak-lsp#694]
I have seen different other issues about this that were closed. I also see this regex in the kak code handling the matching of placeholders.

I did not dig fully into the details, but placeholders are not selected using the LSP server of zig (zls).

Here is my `kak-lsp.toml`.

```
[language.zig]
filetypes = ["zig"]
roots = ["build.zig"]
command = "zls"
```

I also have this mapping in my kak config to iterate through placeholders with `<tab>`.

```
map global insert <tab> '<a-;>:try lsp-snippets-select-next-placeholders catch %{ execute-keys -with-hooks <lt>tab> }<ret>' -docstring 'Select next snippet placeholder'

```

And here is a short snippet of zig code (`|` is my cursor). If I do `<c-n>` until I select `fori`, then hit `<tab>` to try to focus the placeholders, it does not work and simply enter the full snippet plus the tab afterwards.

```
pub fn main() !void {
    for|
}
```

```
pub fn main() !void {
    for ($1, 0..) |${2:value}, ${3:i}| {$0}    |
}
```

![snippet_kak](https://github.com/kak-lsp/kak-lsp/assets/3764103/681b5aff-3bfa-479e-91b0-c78b04a6ea0d)

Not sure what is happening. If anyone has an idea.

**Comment by krobelus** [comment]
Hmm it works for me with `zls` 0.11.0, latest kak-lsp master and Kakoune v2023.07.29
If I type `for<c-n><c-n>` to select `fori`, followed by `1<tab>2<tab>3<tab>4` I get

`for (1, 0..) |2, 3| {4}`

Are you sure you have `snippet_support = true` in your config?
In general it's good to try with an empty config, for example

```sh
HOME=$(mktemp -d) kak -e '
    eval %sh{kak-lsp --kakoune -s $kak_session}
    set global lsp_cmd "kak-lsp -s %val{session} -vvvv --log /tmp/kak-lsp.log"
    map global insert <tab> %{<a-;>:try lsp-snippets-select-next-placeholders catch %{ execute-keys -with-hooks <lt>tab> }<ret>}
    lsp-enable' build.zig
```

if that doesn't work, can you give logs and versions

**Comment by lobre** [comment]
Snippets were effectively disabled in my config. I cannot imagine I spent 2 hours trying to debug that without spotting it ^^.

So I had to add `snippet_support = true` in my config. Maybe it is just me, but I thought that it was activated by default because I read the following from the README:

> Snippet support can be disabled by setting snippet_support = false at the top level of the config.

But it is not. Or maybe it just means that the [default template provided in the repo](https://github.com/kak-lsp/kak-lsp/blob/ae3d052928a482620f62fe3ee866ad072808bac1/kak-lsp.toml#L1) has it enabled, but for people (like me) that started their config from scratch, it is not that clear.

Thanks a lot for your help.
